src/pybind/mgr/smart/module.py

```diff
@@ -1,33 +1,561 @@
-        self.log.error("handle_command")
-
-        if cmd['prefix'] == 'osd smart get':
-            result = CommandResult('')
-            self.send_command(result, 'osd', cmd['osd_id'], json.dumps({
-                'prefix': 'smart',
-                'format': 'json',
-            }), '')
-            r, outb, outs = result.wait()
-            return (r, outb, outs)
+        self.log.info("Handling command: '%s'" % str(cmd))
+
+        if cmd['prefix'] == 'smart status':
+            s = {
+                'active': self.active,
+            }
+            return (0, json.dumps(s, indent=4), '')
+
+        elif cmd['prefix'] == 'smart on':
+            if not self.active:
+                self.set_config('active', '1')
+                self.active = True
+            self.event.set()
+            return (0, '', '')
+
+        elif cmd['prefix'] == 'smart off':
+            if self.active:
+                self.set_config('active', '')
+                self.active = False
+            self.event.set()
+            return (0, '', '')
+
+        elif cmd['prefix'] == "smart osd scrape":
+            # TODO check if cmd['osd_id'] is valid
+            return self.handle_osd_smart_scrape(cmd['osd_id'])
+
+        elif cmd['prefix'] == "smart osd scrape-all":
+            return self.handle_osd_smart_scrape_all()
+
+        elif cmd['prefix'] == "smart osd dump":
+            # TODO check that cmd['obj_name'] is specified and valid
+            return self.handle_osd_smart_dump(cmd)
+
+        elif cmd['prefix'] == "smart osd dump-all":
+            # TODO there might be an issue when running this cmd from cli since
+            # self.last_scrape_time is updated via serve() and
+            # do_osd_smart_dump() might not return results
+            return self.handle_osd_smart_dump_all()
+
+        elif cmd['prefix'] == "osd devices":
+            if "osd_id" not in cmd:
+                return (0, "", "Please specify <osd-id>")
+            osd = self.get_metadata('osd', cmd['osd_id'])
+            if not osd:
+                return ("0", "", "osd %s is not found!" % cmd['osd_id'])
+            return (0, "", osd['devices'])
+
+        elif cmd['prefix'] == "device osds":
+            if "host_device" not in cmd:
+                return (0, "", "Please specify <hostname:device>")
+            osds = self.get("osd_metadata")
+            host, device = cmd['host_device'].split(":")
+            device_osds = []
+            for osd, data in osds.items():
+                if data['hostname'] == host:
+                    # TODO it seems like val here is a string and not a list
+                    # TODO check case of 'host:' (empty set, all osds have it)
+                    if device in data['devices']:
+                        device_osds += osd
+            return (0, "", str(device_osds))
+
+        elif cmd['prefix'] == "smart warn":
+            # TODO check why it's not in 'ceph health' output
+            self.set_health_checks({
```

Comment:
`set_health_checks` expects the complete list of health checks from this module.  So you need to accumulate all health checks that might be active, before calling this function.  You also need to call `set_health_checks({})` when there aren't any checks failing, to clear any that were raised before.